### PR TITLE
Chore: add eslint-plugin-security to detect security issues.

### PR DESCRIPTION
**Pull request by Aladdin-ADD**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Other, please explain: add [eslint-plugin-security](https://github.com/nodesecurity/eslint-plugin-security) to detect security issues.

**What changes did you make? (Give an overview)**

**Is there anything you'd like reviewers to focus on?**

it does provide some useful rules, but seems to be too strict for ESLint codebase(reporting some warnings). thoughts?

### CODE REVIEW

File: tests/lib/cli-engine.js [hunk 1]
```diff
@@ -598,8 +598,6 @@ describe("CLIEngine", () => {
             const report = engine.executeOnFiles(["lib/cli*.js"]);
 
             assert.equal(report.results.length, 2);
-            assert.equal(report.results[0].messages.length, 0);
```
Comment (gyandeeps) [inline tests/lib/cli-engine.js]: We cant't remove these.

**Comment by gyandeeps** [comment]
Do we really need this plugin? Just trying to think.

**Comment by Aladdin-ADD** [comment]
actually I'm not sure. but for some cases ( #9001 ), it is helpful.

File: tests/lib/cli-engine.js (hunk 1 above)
Comment (Aladdin-ADD) [reply tests/lib/cli-engine.js]: this plugin reports some warnings, so `messages.length` is not 0, which cause the CI failing.

Comment (gyandeeps) [reply tests/lib/cli-engine.js]: you can use `[assert.lengthOf](http://chaijs.com/api/assert/#method_lengthof)` function I guess.

**Comment by not-an-aardvark** [comment]
I don't think it would have caught the issue in #9001 anyway -- it only seems to catch regexes that take exponential time, not quadratic time.

File: tests/lib/cli-engine.js (hunk 1 above)
Comment (not-an-aardvark) [reply tests/lib/cli-engine.js]: Personally, I would prefer to not have any warnings be reported when we lint the codebase. Would we be able to fix the warnings?

Comment (Aladdin-ADD) [reply tests/lib/cli-engine.js]: no, it help identify potential security hotspots, but finds a lot of false positives which need triage by a human.

**Pull request closed without merging**